### PR TITLE
chore(docs) Documentation of Amazon SQS-SNS based pubsub

### PR DESCRIPTION
**Pull request by clanesf**
Initial draft of Amazon SQS-SNS pubsub docs. Please give me any feedback!

### CODE REVIEW

**Comment by clanesf** [comment]
@emjburns - Here is the initial draft. I'm still going to go over it again and run through the steps just to make sure I haven't missed anything. When I did it, I handled a lot of permissions and events, etc through the UI, but I wanted to give a way to do it through the AWS CLI instead to have better parity with the way the gcloud pubsub stuff works.

File: setup/triggers/amazon.md [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+    "TopicArn": "arn:aws:sns:us-east-1:1234567890:aws-pubsub-test-topic"
+ }
+```
+* Make a note of the TopicArn. You will need this to configure Echo via the echo-local.yml file to tell Spinnaker to subscribe to this topic.
+
+## Creating an Simple Queue Service (SQS) Queue
+
+The next step is to create an SQS Queue that will receive the notification messages.
+
+* Use the AWS Command Line tool or the AWS Console website to create an SQS Queue that Spinnaker will listen to. You can do this by running the following commands:
```
Comment (emjburns) [inline setup/triggers/amazon.md]: Actually, spinnaker will do this for you. You just need to put what you'd like the arn to be.

File: setup/triggers/amazon.md [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+EOF
+ bash4.4$ aws s3api put-bucket-notification-configuration --bucket spin-pubsub-test-bucket \
+ --notification-configuration file:///tmp/spin-pubsub-notification.json    
+ bash4.4$
+```
+* Congratulations! You have now done all of the configuration in AWS to send pubsub messages. All we have to do now is configure Spinnaker to subscribe to and retrieve these messages.
+
+## Writing your Echo configuration
+
+Amazon pubsub configuration is not yet supported via Halyard, so you'll need to define the details regarding your subscription in an echo-local.yml file. Put this echo-local.yml file in `~/.hal/<Deployment Profile>/profiles/` directory and Halyard will automatically deploy it alongside echo the next time you run `hal deploy apply`. Halyard's default profile is named `default`, so if you haven't set up a different profile, this `echo-local.yaml` file would go in the `~/.hal/default/profiles/` directory on your halyard instance.
```
Comment (emjburns) [inline setup/triggers/amazon.md:113]: Can we note that PRs for halyard support would be a decent place to start for someone looking to help out with the code?

**Review by emjburns — APPROVED** [review]
This is great! I'd love for you to link the actual AWS docs that you referenced for each thing (create SNS topic, etc) so that it's easy to keep this up to date.

Thanks so much for this documentation!

**Comment by emjburns** [comment]
@dorbin might also have some good feedback for you :)

When you're ready for this to be merged ping me and I'll take care of it.

File: setup/triggers/amazon.md (hunk 1 above)
Comment (clanesf) [reply setup/triggers/amazon.md]: Okay, so I've just gone back through and tested all these steps and confirmed that they work in this order. If you just specify the ARN for the queue in the config file, does it automatically handle the permissions to subscribe you to the topic you provide?

File: setup/triggers/amazon.md (hunk 2 above)
Comment (clanesf) [reply setup/triggers/amazon.md:113]: Done.

File: setup/triggers/amazon.md (hunk 1 above)
Comment (emjburns) [reply setup/triggers/amazon.md]: Yes it should - as long as you actually have the aws permissions to do the subscribe

**Comment by clanesf** [comment]
@emjburns - Okay, I put in a notice about queue creation being automatic and to skip that section if they want. I figured I wouldn't remove it just in case people want to do it manually or have trouble having one automatically created maybe this will help them figure out potential permissions issues.

Please merge this if you feel it's good enough now.

**Comment by dorbin** [comment]
@clanesf 
Quick comment before I do a more thorough review: can you add an entry to the left-nav toc ( /_data/navigation.yml.

**Pull request merged**